Makefile

```diff
@@ -1,54 +1,52 @@
-.PHONY: kbs
-kbs:
-	cargo build
+AS_TYPE ?= coco-as
+HTTPS_CRYPTO ?= rustls
+POLICY_ENGINE ?=
 
-.PHONY: kbs-coco-as
-kbs-coco-as:
-	cargo build --no-default-features --features coco-as-builtin,rustls
+COCO_AS_INTEGRATE_TYPE ?= builtin
 
-.PHONY: kbs-coco-as-grpc
-kbs-coco-as-grpc:
-	cargo build --no-default-features --features coco-as-grpc,rustls
+INSTALL_DESTDIR ?= /usr/local/bin
 
-.PHONY: kbs-coco-as-no-verifier
-kbs-coco-as-no-verifier:
-	cargo build --no-default-features --features coco-as-builtin-no-verifier,rustls
+ifeq ($(AS_TYPE), coco-as)
+  AS_FEATURE = $(AS_TYPE)-$(COCO_AS_INTEGRATE_TYPE)
+else
+  AS_FEATURE = $(AS_TYPE)
+endif
 
-.PHONY: kbs-coco-as-openssl
-kbs-coco-as-openssl:
-	cargo build --no-default-features --features coco-as-builtin,openssl
+build: background-check-kbs
 
-.PHONY: kbs-coco-as-grpc-openssl
-kbs-coco-as-grpc-openssl:
-	cargo build --no-default-features --features coco-as-grpc,openssl
+.PHONY: background-check-kbs
+background-check-kbs:
+	cargo build --release --no-default-features --features $(AS_FEATURE),resource,$(HTTPS_CRYPTO),$(POLICY_ENGINE)
 
-.PHONY: kbs-coco-as-no-verifier-openssl
-kbs-coco-as-no-verifier-openssl:
-	cargo build --no-default-features --features coco-as-builtin-no-verifier,openssl
+.PHONY: passport-abs
+passport-abs:
```

Comment:
I wonder if it is really worth distinguishing between the KBS and the ABS. It looks like they are basically the same thing. I know that the RATS diagrams show the passport model making a direct connection between Attester and Verifier, but we aren't going to do that either way, so why not just say that we always have the KBS in front of the AS? I think introducing this new piece might confuse people. Instead we can just say that if you are using the passport model you can build the KBS without support for resources and then you can use another KBS that does have support for resources.